Renamer/IconChanger.cs

```diff
@@ -0,0 +1,156 @@
+            FieldInfo[] scrolls = typeof(KSP.UI.Screens.AstronautComplex).GetFields(BindingFlags.NonPublic | BindingFlags.Instance).Where(c => c.FieldType == typeof(KSP.UI.UIList)).ToArray();
+
+            for (int i = 0; i < scrolls.Length; i++)
+            {
+                scroll = (KSP.UI.UIList)scrolls[i].GetValue(complex);
+                for (int j = 0; j < scroll.Count; j++)
+                {
+                    KSP.UI.UIListItem listItem = scroll.GetUilistItemAt(j);
+                    cic = listItem.GetComponent<KSP.UI.CrewListItem>();
+                    cic.AddButtonInputDelegate(new UnityAction<KSP.UI.CrewListItem.ButtonTypes, KSP.UI.CrewListItem>(RebuildAstronautComplex));
+                    changeKerbalIcon(cic);
+                }
+            }
+        }
+
+        public void BuildCrewAssignmentDialogue()
+        {
+            if ((object)KSP.UI.CrewAssignmentDialog.Instance == null)
+            {
+                return;
+            }
+
+            KSP.UI.CrewAssignmentDialog dialogue = KSP.UI.CrewAssignmentDialog.Instance;
+            KSP.UI.CrewListItem cic;
+
+            for (int j = 0; j < dialogue.scrollListAvail.Count; j++)
+            {
+                KSP.UI.UIListItem listItem = dialogue.scrollListAvail.GetUilistItemAt(j);
+                cic = listItem.GetComponent<KSP.UI.CrewListItem>();
+                cic.AddButtonInputDelegate(new UnityAction<KSP.UI.CrewListItem.ButtonTypes, KSP.UI.CrewListItem>(RebuildCrewAssignmentDialogue));
+                changeKerbalIcon(cic);
+            }
+            for (int j = 0; j < dialogue.scrollListCrew.Count; j++)
+            {
+                KSP.UI.UIListItem listItem = dialogue.scrollListCrew.GetUilistItemAt(j);
+                cic = listItem.GetComponent<KSP.UI.CrewListItem>();
+                if ((object)cic != null)
+                {
+                    cic.AddButtonInputDelegate(new UnityAction<KSP.UI.CrewListItem.ButtonTypes, KSP.UI.CrewListItem>(RebuildCrewAssignmentDialogue));
+                    changeKerbalIcon(cic);
+                }
+            }
+        }
+
+        public void RebuildAstronautComplex(KSP.UI.CrewListItem.ButtonTypes type, KSP.UI.CrewListItem cic)
+        {
+            StartCoroutine(CallbackUtil.DelayedCallback(1, BuildAstronautComplex));
+        }
+
+        public void RebuildCrewAssignmentDialogue(KSP.UI.CrewListItem.ButtonTypes type, KSP.UI.CrewListItem cic)
+        {
+            StartCoroutine(CallbackUtil.DelayedCallback(1, BuildCrewAssignmentDialogue));
+        }
+
+        private void changeKerbalIcon(KSP.UI.CrewListItem cic)
+        {
+            if ((object)cic.GetCrewRef() == null)
+            {
+                return;
+            }
+
+            ConfigNode data = null;
+            foreach (ConfigNode node in GameDatabase.Instance.GetConfigNodes("KERBALRENAMER"))
+            {
+                data = node;
```

Comment:
are you doing this just to get the last instance of the KERBALRENAMER config node?  What about the other instances?  Should there only be one instance?

Reply:
This isn't my code but I assume the intent is that there should only be one KERBALRENAMER config node and if there are multiple nodes the last one wins.